Lib/urllib/parse.py

```diff
@@ -38,13 +39,25 @@
            "urlsplit", "urlunsplit", "urlencode", "parse_qs",
            "parse_qsl", "quote", "quote_plus", "quote_from_bytes",
            "unquote", "unquote_plus", "unquote_to_bytes",
-           "DefragResult", "ParseResult", "SplitResult",
+           "DefragResult", "ParseResult", "SplitResult", "SchemeFlag",
            "DefragResultBytes", "ParseResultBytes", "SplitResultBytes"]
 
 # A classification of schemes.
 # The empty string classifies URLs with no scheme specified,
 # being the default value returned by “urlsplit” and “urlparse”.
 
+class SchemeFlag(Flag):
+    """SchemeFlag is an enum with the members RELATIVE, NETLOC, and
+    PARAMS. These describe methods for URL resolution, usually by
+    scheme. These resolution classes determine, namely, whether a
+    scheme supports, respectively, relative addressing, preserving the
+    netloc (domain name), and preserving the parameters.
+    """
+    NONE = 0
```

Comment:
We don't need a `NONE` flag.  `SchemeFlag()` or `SchemeFlag(0)` (etc.) will return a zero-valued unnamed flag, which is fine.